clippy_utils/src/higher.rs

```diff
@@ -557,13 +559,15 @@ impl FormatArgsExpn<'tcx> {
                     _ => None,
                 })
                 .collect();
+            if let PatKind::Tuple(arg_names, None) = arm.pat.kind;
```

Comment:
This is okay for now since it fixes an ICE, but note that we want to move away from depending on the exact HIR per #7843.